### PR TITLE
#119 TsLogged.java:38-42: This class not implemented yet

### PR DESCRIPTION
**Pull request by dmzaytsev**
task #119 
implemented TsLogged 
created TsLogedTest
created stub for BkLogged

### CODE REVIEW

**Comment by davvd** [comment]
Thanks for the pull request, let me find a reviewer..

**Comment by davvd** [comment]
@pinaf it's yours,please review

File: src/main/java/org/takes/facets/slf4j/BkLogged.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
```
Comment (pinaf) [inline src/main/java/org/takes/facets/slf4j/BkLogged.java]: @dmzaytsev let's remove this blank line.

File: src/main/java/org/takes/facets/slf4j/BkLogged.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+ *
+ * <p>The class is immutable and thread-safe.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @todo #28:30min/DEV This class not implemented yet, but has to be.
+ *  Please implement it like TkLogged and don't forget about unit tests.
+ *  In addition, need to be implement classes FtLogged, PsLogged
+ *  which will make loggable interfaces Front, Pass. They should all be
+ *  implemented also like TkLogged.
+ * @since 0.11.2
```
Comment (pinaf) [inline src/main/java/org/takes/facets/slf4j/BkLogged.java]: @dmzaytsev 0.11.2 was already released. maybe 0.11.3 ?

File: src/main/java/org/takes/facets/slf4j/TsLogged.java [hunk 3]
```diff
@@ -25,6 +25,7 @@
 package org.takes.facets.slf4j;
 
 import java.io.IOException;
+import lombok.EqualsAndHashCode;
```
Comment (pinaf) [inline src/main/java/org/takes/facets/slf4j/TsLogged.java]: @dmzaytsev let's remove the blank line between the copyright and `package` here also.

File: src/main/java/org/takes/facets/slf4j/TsLogged.java [hunk 4]
```diff
@@ -35,17 +36,61 @@
...
+     *  it don't has this feature.
+     *  See details here https://github.com/yegor256/takes/issues/101
+     *  We will use toConsole() in this way
+     *  this.target.log("...", this.origin.toConsole(), request.toConsole, ...)
+     */
     @Override
     public Take route(final Request request) throws IOException {
-        throw new UnsupportedOperationException("not implemented yet");
+        final long started = System.currentTimeMillis();
+        final Take take = this.origin.route(request);
```
Comment (pinaf) [inline src/main/java/org/takes/facets/slf4j/TsLogged.java]: @dmzaytsev can't we inline this variable?

File: src/test/java/org/takes/facets/slf4j/TsLoggedTest.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
```
Comment (pinaf) [inline src/test/java/org/takes/facets/slf4j/TsLoggedTest.java]: @dmzaytsev let's remove this blank line.

File: src/test/java/org/takes/facets/slf4j/TsLoggedTest.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import org.takes.Take;
+import org.takes.Takes;
+import org.takes.rq.RqFake;
+import org.takes.ts.TsEmpty;
+
+/**
+ * Test case for {@link TsLogged}.
+ * @author Dmitry Zaytsev (dmitry.zaytsev@gmail.com)
+ * @version $Id$
+ * @since 0.11.2
```
Comment (pinaf) [inline src/test/java/org/takes/facets/slf4j/TsLoggedTest.java]: @dmzaytsev same comment here on release.

**Comment by pinaf** [comment]
@dmzaytsev 5 comments above.

**Comment by dmzaytsev** [comment]
@pinaf I did push new commit, but it doesn't appear here.
[here](https://github.com/yegor256/takes/compare/master...dmzaytsev:119) I see two commits in the PR #122, but [here](https://github.com/yegor256/takes/pull/122/commits) I see only one commit

**Comment by dmzaytsev** [comment]
@pinaf I'll contact with Github support

**Comment by dmzaytsev** [comment]
@pinaf response from support
>Sorry for the trouble. We deployed a change that seems to have created some PR buginess.
We're working on a fix, and I'll let you know when we have an update.

**Comment by dmzaytsev** [comment]
@pinaf  Github resolved the issue
the PR updated

**Comment by pinaf** [comment]
@dmzaytsev looks good! thanks!

**Comment by pinaf** [comment]
@rultor merge

**Comment by yegor256** [comment]
@dmzaytsev `Takes` interface was removed from the project (in #133), together with this puzzle. you can just close this pull request and the original ticket will be closed automatically (and paid)

**Comment by dmzaytsev** [comment]
@yegor256 if this puzzle will be closed, how should be implemented BkLogged, FtLogged, PsLogged ?

**Comment by yegor256** [comment]
@dmzaytsev good question. We'll have to create tickets for them, separately, not through puzzles

**Comment by davvd** [comment]
@elenavolokhova review this ticket please, for compliance with our [quality rules](http://www.teamed.io/qa.html)

**Comment by elenavolokhova** [comment]
@pinaf 
>davvd commented 6 days ago
Thanks for the pull request, let me find a reviewer..

 According to our [quality rules](http://www.teamed.io/qa.html):
 > Messages in a ticket always start with a name of a user they are addressed to.
 
 As far as you are the performer of this ticket you are responsible for its quality. Please make sure that everyone follows this rule in your tickets in the future.
 
 Please confirm that everything is clear and you have no questions regarding this approach.

**Comment by pinaf** [comment]
@yegor256 this was closed 5 days ago

**Comment by yegor256** [comment]
@pinaf same here, waiting for QA confirmation here, see http://www.teamed.io/qa.html

**Comment by pinaf** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@pinaf Thank you!

**Comment by elenavolokhova** [comment]
@davvd Ticket passed QA review. Quality is acceptable.

**Comment by davvd** [comment]
> @davvd Ticket passed QA review. Quality is acceptable.

@elenavolokhova thanks, got it. everybody, please try to make it good next time

**Comment by davvd** [comment]
@pinaf I added 10 mins to @elenavolokhova (for QA review) in transaction 55485753. Much obliged! I have added **20 mins** to your account in payment "AP-2ES826934D191343A", 49 hours and 12 mins spent. extra minutes for review comments (c=5). +20 added to your rating, current score is: [+7004](http://www.netbout.com/b/35128?open=rating)

**Pull request closed without merging**